network/builder/src/builder.rs

```diff
@@ -225,11 +229,45 @@ impl NetworkBuilder {
             config.mutual_authentication,
         );
 
-        match &config.discovery_method {
-            DiscoveryMethod::Onchain => {
-                network_builder.add_validator_set_listener(pubkey, config.encryptor());
+        let discovery_methods: Vec<_> = config
+            .discovery_methods
+            .iter()
+            .filter(|method| &&DiscoveryMethod::None != method)
+            .collect();
+
+        // TODO: This is a backwards compatibility feature.  Deprecate discovery_method
+        if config.discovery_method != DiscoveryMethod::None && !discovery_methods.is_empty() {
+            panic!("Can't specify discovery_method and discovery_methods")
```

Comment:
should ideally be addressed upstream and config should have a method that retrieves this -- maybe?